### PR TITLE
Add api endpoint for starting collection syncs.

### PR DESCRIPTION
**Pull request by newswangerd**
Add a `/api/automation-hub/content/<distribution>/v3/sync/` endpoint that allows users to start syncs on the collection remote attached to the specified distribution.

Syncs are kicked off with a `POST` request and return the ID of the task.

Response
```json
{
    "task": "3024deed-4df2-44c7-a3f1-a38124c6ccf7"
}
```

### CODE REVIEW

File: galaxy_ng/app/api/v3/urls.py [hunk 1]
```diff
@@ -18,6 +18,11 @@
 
 
 urlpatterns = [
+    path(
```
Comment (rochacbruno) [inline galaxy_ng/app/api/v3/urls.py:21]: I suggest to map urls under `content/` that way it doesn’t get included when the route doesn’t have the `content/<distro>` - see #410 

https://github.com/ansible/galaxy_ng/blob/e5694720482a7a65266ee83c0054550c515dca47/galaxy_ng/app/api/urls.py#L38

File: galaxy_ng/app/api/v3/views/sync.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+        remote = distro.repository.remote
+
+        # check if the user is pulling from api v2 (galaxy.ansible.com)
+        # if no requirements file is set, prevent the sync from happening
+        if 'api/v2' in remote.url and not remote.requirements_file:
+            raise ValidationError(
+                detail={
+                    'requirements_file':
+                        'Syncing content from galaxy.ansible.com without specifying a'
```
Comment (rochacbruno) [inline galaxy_ng/app/api/v3/views/sync.py:39]:
```suggestion
                        'Syncing content from galaxy.ansible.com without specifying a '
```

missing a whitespace at the end

File: galaxy_ng/app/api/v3/views/sync.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    def post(self, request: Request, *args, **kwargs) -> Response:
+        distro_path = kwargs['path']
+        distro = get_object_or_404(pulp_models.AnsibleDistribution, base_path=distro_path)
+
+        if not distro.repository or not distro.repository.remote:
+            raise ValidationError(
+                detail={'remote': f'The {distro_path} distribution does not have'
+                                  ' any remotes associated with it.'})
+
+        remote = distro.repository.remote
```
Comment (rochacbruno) [inline galaxy_ng/app/api/v3/views/sync.py:31]: `remote` doesn't have a `requirement_file` so you need to get the `CollectionRemote`object.
```suggestion
        remote = distro.repository.remote.ansible_collectionremote
```

File: galaxy_ng/app/access_control/statements/standalone.py [hunk 4]
```diff
@@ -58,6 +58,12 @@
             "action": ["list"],
             "principal": "authenticated",
             "effect": "allow"
+        },
+        {
+            "action": ["sync"],
+            "principal": "authenticated",
+            "effect": "allow",
+            "condition": "has_model_perms:ansible.update_ansible_remote"
```
Comment (rochacbruno) [inline galaxy_ng/app/access_control/statements/standalone.py:66]:
```suggestion
            "condition": "has_model_perms:ansible.change_collectionremote"
```

**Comment by rochacbruno** [comment]
This changes have been merged to #410 so we can close this PR.

**Comment by newswangerd** [comment]
This got added in #410

**Pull request closed without merging**